src/components/GameBoard.tsx

```diff
@@ -0,0 +1,31 @@
+import { GameBoardState, SquareState } from '@/shared/types';
+import styles from './GameBoard.module.css'
+
+interface GameBoardProps {
+  grid2D: GameBoardState;
+  renderSquare?: (tile: SquareState) => JSX.Element;
+}
+
+const GameBoard = ({ grid2D, renderSquare }: GameBoardProps) => {
+  return (
+    <div className={styles.grid}>
+      {grid2D.map((row, rowIndex) => row.map((cell, cellIndex) => (
+          <div key={`${rowIndex}${cellIndex}`} className={styles.cell}>
+            {renderSquare ? renderSquare(cell) : defaultSquareRenderer(cell)}
+          </div>
+        ))
+      )}
+    </div>
+  );
+}
+
+export const defaultSquareRenderer = (tile: SquareState) => {
```

Comment:
Just a note 📓  This would need to render actionable/clickable squares in the future probably.